source/common/network/utility.h

```diff
@@ -78,6 +78,24 @@ class Utility {
    */
   static uint32_t portFromTcpUrl(const std::string& url);
 
+  /*
+  * Parse an internet host address (IPv4 or IPv6) and create an Instance from it. Throw
+  * an exception if unable to parse the address. The address must not include a port number.
+  * @param ip_address string to be parsed as an internet address.
+  * @return pointer to the Instance, or nullptr if unable to parse the address.
+  */
+  static Address::InstanceConstSharedPtr parseInternetAddress(const std::string& ip_address);
+
+  /*
+  * Parse an internet host address (IPv4 or IPv6) AND port, and create an Instance from it.
+  * Throw an exception if unable to parse the address.
```

Comment:
You could also use an `@throws` Doxygen annotation, but this isn't common in the code base.